File: 0167_Two_Sum_II-Input_array_is_sorted.py
class Solution:
    def twoSum(self, numbers: List[int], target: int) -> List[int]:
        d = collections.Counter(numbers)
        for i, el in enumerate(numbers):
            diff = target - el
            if d[diff] >= 1:
                nums = numbers[i+1:]
                j = nums.index(diff)
                return [i+1, i+1+j+1]
        return None
            

File: 0167_Two_Sum_II-Input_Array_Is_Sorted/0167_Two_Sum_II-Input_Array_Is_Sorted.java
import java.util.HashMap;
import java.util.Arrays;
class Solution {
    public int[] twoSum(int[] numbers, int target) {
        int complement;
        int len = numbers.length;
        HashMap<Integer, Integer> map = new HashMap<Integer, Integer>();
        for (int i=0;i<len;i++) {
            complement = target - numbers[i];
            if (map.containsKey(complement)) {
                int[] ans = {i+1, map.get(complement)+1};
                Arrays.sort(ans);
                return ans;
            }
            map.put(numbers[i], i);
        }
        throw new IllegalStateException();
    }
}